docs/privatelink.md

```diff
@@ -0,0 +1,276 @@
+    - compute.forwardingRules.pscCreate
+    - compute.forwardingRules.pscDelete
+    - compute.networks.use
+    - compute.regionOperations.get
+    - compute.subnetworks.use
+    - servicedirectory.namespaces.create
+    - servicedirectory.services.create
+    - servicedirectory.services.delete
+
+
+### Configure the GCP endpointVPCInventory
+
+The Link Actuator needs to know where to create the cluster's Endpoint. This
+can be configured in the `spec.privateLink.gcp.endpointVPCInventory` parameter
+in the hiveconfig. When creating the endpoint, the actuator will choose the
+subnet from this list that is in the same region and has the least number of
+addresses provisioned in the cloud provider.
+
+1. Create a network for the PrivateLink Controller to use when creating
+endpoints.
+
+1. Create one or more subnets in the network for each region that private
+clusters will be created in.
+
+1. Make sure all the Hive environments (Hive VPCs) have network reachability to
+these subnets using peering, transit gateways, VPNs, etc.
+
+1. Update the HiveConfig to enable these subnets.
+
+```yaml
+## hiveconfig
+spec:
+  privateLink:
+    gcp:
+      endpointVPCInventory:
+      - network: network-1
+        subnets:
+        - region: us-east1
+          subnet: subnet1
+        - region: us-east2
+          subnet: subnet2
+```
+
+## AWS Deploying a PrivateLink cluster
+
+Support for AWS private clusters is still managed by the origional [AWS
+PrivateLink Controller](awsprivatelink.md). The goal is to eventually merge its
+functionality into this PrivateLink Controller.
+
+## GCP Deploying a PrivateLink cluster
+
+Once Hive has been configured to support PrivateLink GCP clusters, you can
+deploy a cluster by setting `privateServiceConnect.enabled` to true. This is
+only supported in regions where Hive has been configured with at least one
+subnet in [endpointVPCInventory](#configure-the-gcp-endpointvpcinventory)
+with the same region. The Link Actuator uses the service account specified in
+the clusterDeployment when managing the Service Attachment, Subnet, and
+Firewall resources.
+
+
+```yaml
+## clusterDeployment
+spec:
+  platform:
+    gcp:
+      privateServiceConnect:
+        enabled: true
+```
+
+By default, the Link Actuator will create a subnet and associated firewall in
+the clusters's VPC to be used when creating the Service Attachment. The default
+CIDR of the subnet can be overriden with the `serviceAttachment.subnet.cidr`
+parameter
+
+```yaml
+## clusterDeployment
+spec:
+  platform:
+    gcp:
+      privateServiceConnect:
+        serviceAttachment:
+          subnet:
+            cidr: 192.168.1.0/30
+```
+
+Alternatively, a pre-existing subnet can be specified and the Link Actuator
+will use it instead of creating a new one. This subnet must have a purpose of
+`PRIVATE_SERVICE_CONNECT` and have routing and firewall rule(s) enabling access
+to the int-api load balancer. Specifying an existing subnet is required when
```

Comment:
Is it called `int-api` or `api-int`?

Reply:
Technically, api-internal
[Fixed](https://github.com/openshift/hive/compare/19933d189d2166ec5fd4733e83297f6d742652b8..3df83c8128307810f05ed8391f1c67eaa57c0ff8)